streampipes-processors-transformation-jvm/src/main/java/org/apache/streampipes/processors/transformation/jvm/processor/booloperator/logical/BooleanOperatorProcessor.java

```diff
@@ -0,0 +1,99 @@
+                .withAssets(Assets.DOCUMENTATION, Assets.ICON)
+                .withLocales(Locales.EN)
+                .category(DataProcessorType.ENRICH)
+                .requiredStream(
+                        StreamRequirementsBuilder
+                                .create()
+                                .requiredProperty(EpRequirements.anyProperty())
+                                .build())
+                .requiredTextParameter(Labels.withId(BOOLEAN_PROCESSOR_INPUT_KEY))
+                .outputStrategy(OutputStrategies.append(
+                        PrimitivePropertyBuilder.create(
+                                Datatypes.String, BOOLEAN_PROCESSOR_OUT_KEY)
+                                .build()))
+                .build();
+    }
+
+    @Override
+    public void onInvocation(ProcessorParams processorParams, SpOutputCollector spOutputCollector, EventProcessorRuntimeContext eventProcessorRuntimeContext) throws SpRuntimeException {
+        BooleanOperationInputConfigs configs = processorParams.extractor().singleValueParameter(BOOLEAN_PROCESSOR_INPUT_KEY, BooleanOperationInputConfigs.class);
+        preChecks(configs);
+        this.configs = configs;
+    }
+
+    @Override
+    public void onEvent(Event event, SpOutputCollector spOutputCollector) throws SpRuntimeException {
+        List<String> properties = configs.getProperties();
+        BooleanOperatorType operatorType = configs.getOperator();
+        Boolean firstProperty = event.getFieldBySelector(properties.get(0)).getAsPrimitive().getAsBoolean();
+        IBoolOperation<Boolean> boolOperation = BoolOperationFactory.getBoolOperation(operatorType);
+        Boolean result;
+        if (properties.size() == 1) {
+            // support for NOT operator
+            result = boolOperation.evaluate(firstProperty, firstProperty);
+
+        } else {
+            Boolean secondProperty = event.getFieldBySelector(properties.get(1)).getAsPrimitive().getAsBoolean();
+            result = boolOperation.evaluate(firstProperty, secondProperty);
+
+            //loop through rest of the properties to get final result
+            for (int i = 2; i < properties.size(); i++) {
+                result = boolOperation.evaluate(result, event.getFieldBySelector(properties.get(i)).getAsPrimitive().getAsBoolean());
+            }
+
+        }
+        event.addField(BOOLEAN_PROCESSOR_OUT_KEY, result);
+        spOutputCollector.collect(event);
+    }
+
+    @Override
+    public void onDetach() throws SpRuntimeException {
+        configs = null;
+    }
+
+    private void preChecks(BooleanOperationInputConfigs configs) {
+        BooleanOperatorType operatorType = configs.getOperator();
+        List<String> properties =  configs.getProperties();
+        if (operatorType == NOT) {
+            assert properties.size() == 1 : "NOT operator can operate only on single operand";
```

Comment:
Shall we throw a `SpRuntimeException` instead of `assert`?

Reply:
Sure. I'll do the same.